Kernel/Devices/SerialDevice.cpp

```diff
@@ -7,71 +7,29 @@
 
 #include <Kernel/Devices/SerialDevice.h>
 #include <Kernel/IO.h>
+#include <Kernel/TTY/ConsoleManagement.h>
+#include <Kernel/TTY/TTY.h>
 
 namespace Kernel {
 
 UNMAP_AFTER_INIT SerialDevice::SerialDevice(IOAddress base_addr, unsigned minor)
-    : CharacterDevice(4, minor)
+    : TTY(4, minor)
+    , IRQHandler((base_addr == (IOAddress)SERIAL_COM1_ADDR || base_addr == (IOAddress)SERIAL_COM3_ADDR) ? 4 : 3)
```

Comment:
This is very hacky, and doesnt handle:
A) COM2 and COM4
B) PCI SerialDevice instances
A better way to handle this should probably be accepting the interrupt number as a constructor parameter (similarly to base_addr)

Reply:
Okay. I don't have any hardware to test it on, though. Thank you for the great suggestions by the way :)

Reply:
Check my pull request that adds support for qemu's emulated pci serial, you can use that to test it.